packages/apps/plugins/plugin-thread/src/ThreadPlugin.tsx

```diff
@@ -221,6 +223,25 @@ export const ThreadPlugin = (): PluginDefinition<ThreadPluginProvides> => {
               state.focus = intent.data?.focus;
               return { data: true };
             }
+
+            case ThreadAction.DELETE: {
+              const { document: doc, thread, cursor } = intent.data ?? {};
+              if (!isDocument(doc) || !isThread(thread)) {
+                return;
+              }
+
+              if (intent.inverse && typeof cursor === 'string') {
+                doc.comments.push({ thread, cursor });
+                return { data: true };
+              } else {
+                const index = doc.comments.findIndex((comment) => comment.thread?.id === thread.id);
```

Comment:
@dmaretskyi should we actively delete the object and not just splice it (and restore above)?

Reply:
@richburdon note that comments are not standalone objects at the moment but apart of the the `Document` schema. Only `Thread`s are separate objects.